File: Práctica 1/Ejercicio1/FactoriaMontana.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package java;

/**
 *
 * @author Usuario
 */
import prac1.FactoriaCarreraBicicleta;
import prac1.bicicletaMontana;
import prac1.carreraMontana;
import java.util.ArrayList;
public class FactoriaMontana implements FactoriaCarreraBicicleta{
    
    
    @Override
    public bicicleta crearBicicleta(int i){
            return new bicicletaMontana(i);       
    }
    
    @Override
    public carrera crearCarrera(ArrayList<bicicleta> participantes){
        return new carreraMontana(participantes);
        
    }
        
}
  

File: Práctica 1/Ejercicio2/FactoriaMontana.py
from FactoriaCarreraBicicleta import FactoriaCarreraBicicleta
from CarreraMontana import CarreraMontana
from BicicletaMontana import BicicletaMontana

class FactoriaMontana(FactoriaCarreraBicicleta):

    def crear_carrera(self, bicicletas, porcentaje):
        return CarreraMontana(bicicletas, porcentaje)

    def crear_bicicleta(self, id):
        return BicicletaMontana(id)